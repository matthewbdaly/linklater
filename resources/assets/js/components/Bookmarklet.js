import React, { Component } from 'react';
import ReactDOM from 'react-dom';

export default class Bookmarklet extends Component {
    render() {
        let url = `javascript:location.href='${this.props.createUrl}?url='+encodeURIComponent(location.href)`;
        return (
            <p className="bookmarklet">
                <a href={url}>LinkLater This!</a>
            </p>
        );
    }
}
