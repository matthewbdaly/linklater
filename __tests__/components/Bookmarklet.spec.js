import React from 'react';
import {findDOMNode} from 'react-dom';
import Bookmarklet from '../../resources/assets/js/components/Bookmarklet';
import renderer from 'react-test-renderer';

describe('Bookmarklet', () => {
  it('renders the bookmarklet', () => {
    const tree = renderer
      .create(<Bookmarklet />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
