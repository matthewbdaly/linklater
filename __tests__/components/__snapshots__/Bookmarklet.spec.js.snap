// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookmarklet renders the bookmarklet 1`] = `
<p
  className="bookmarklet"
>
  <a
    href="javascript:location.href='undefined?url='+encodeURIComponent(location.href)"
  >
    LinkLater This!
  </a>
</p>
`;
